nixos/modules/services/misc/pinnwand.nix

```diff
@@ -0,0 +1,62 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.pinnwand;
+
+  configFile = pkgs.runCommand "pinnwand.toml" {
+    buildInputs = [ pkgs.remarshal ];
+    preferLocalBuild = true;
+  } ''
+    remarshal -if json -of toml \
+    < ${pkgs.writeText "pinnwand.toml" (builtins.toJSON cfg.config)} \
+    > $out
+  '';
+in
+{
+  options.services.pinnwand = {
+    enable = mkEnableOption "Pinnwand";
+
+    port = mkOption {
+      type = types.int;
+      description = "The port to listen on";
+      default = 8000;
+    };
+
+    config = mkOption {
+      type = types.uniq types.attrs;
+      description = ''
+        Your <filename>pinnwand.toml</filename> as a Nix attribute set. Look up
+        possible options in the <link xlink:href="https://github.com/supakeen/pinnwand/blob/master/pinnwand.toml-example">pinnwand.toml-example</link>.
+      '';
+      default = {
```

Comment:
By setting the default here won't all values get wiped out if the user provides any value?

Reply:
The values given are the default values and the config file is merged with the default config, so no option is lost.